lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -119,6 +119,25 @@ static StringRef getNameForObjC(const ValueDecl *VD,
   return VD->getName().str();
 }
 
+static Optional<StringRef>
+printSwiftEnumElemNameInObjC(const EnumElementDecl *EL, llvm::raw_ostream &OS,
```

Comment:
Rather than returning an `Optional<StringRef>`, shouldn't it be sufficient to print a bool saying whether there was a custom name? Alternately, leave the custom name checking separate (as it was before) and only use this to print default names. Then you don't have to return anything.